### ISSUE TITLE
Interop for UWP app exes

### ISSUE THREAD

**Issue by 0xabu** [microsoft/WSL#7303]
### Windows Build Number

10.0.19043.1165

### WSL Version

- [ ] WSL 2
- [X] WSL 1

### Repro Steps

UWP apps such as Python or Windows Terminal can install .exe aliases in `%LOCALAPPDATA%\Microsoft\WindowsApps`, which are implemented under the covers by some reparse magic. Unfortunately (probably because the files are zero bytes in size) these are not usable from WSL:

From CMD:
```
C:\Users\me>AppData\Local\Microsoft\WindowsApps\python.exe -V
Python 3.9.6
```

From WSL (interop enabled):
```
$ cd /mnt/c/Users/me
$ AppData/Local/Microsoft/WindowsApps/python.exe -V
$ echo $?
0
```

The only workaround I have found is to have CMD do it:
```
$ cmd.exe /c python.exe -V
Python 3.9.6
$ cmd.exe /c wt.exe
```
... is there a better way?

(I'm somewhat surprised not to find any existing issues on this topic. Forgive me if I missed something!)

### Expected Behavior

Running any .exe in my PATH invokes the corresponding application, even if it happens to be a shim to an app package.

### Actual Behavior

The command does nothing but returns success.

### Diagnostic Logs

Looking at strace output, it's apparent that exec fails (ENOEXEC), which causes the shell to assume it's a script:
```
$ strace -f sh -c AppData/Local/Microsoft/WindowsApps/python.exe
...
clone(strace: Process 1783 attached
child_stack=NULL, flags=CLONE_CHILD_CLEARTID|CLONE_CHILD_SETTID|SIGCHLD, child_tidptr=0x7f5587e01810) = 1783
[pid  1783] execve("AppData/Local/Microsoft/WindowsApps/python.exe", ["AppData/Local/Microsoft/WindowsA"...], 0x7f558821fb70 /* 31 vars */ <unfinished ...>
[pid  1782] wait4(-1,  <unfinished ...>
[pid  1783] <... execve resumed> )      = -1 ENOEXEC (Exec format error)
[pid  1783] openat(AT_FDCWD, "AppData/Local/Microsoft/WindowsApps/python.exe", O_RDONLY|O_NOCTTY) = 3
[pid  1783] read(3, "", 128)            = 0
[pid  1783] close(3)                    = 0
[pid  1783] execve("/bin/sh", ["/bin/sh", "AppData/Local/Microsoft/WindowsA"...], 0x7f558821fb70 /* 31 vars */) = 0
```

**Comment by 0xabu** [comment]
Slightly better workaround to skip CMD:
```
$ /init AppData/Local/Microsoft/WindowsApps/python.exe -V
Python 3.9.6
```

**Comment by benhillis** [comment]
This is an annoying one that I've tried to fix a couple of different times.  The problem is that python.exe is an app execution alias, and not a real executable (take a look at the size, it shows up as zero).  This means when the Linux kernel tries to read the header for our binfmt interpreter it gets zero bytes.  In the past I tried to fix this by switching our app execution alias logic to also respect the .exe file extension, but it turns out some Linux frameworks (I'm forgetting which) use the .exe extension for their ELF images, so that broke them.

In my opinion this is a bug in the Windows app execution alias logic - I can follow up with them and see what they think.
